### ISSUE TITLE
Support opting-in to libc even on x86_64 etc.

### ISSUE THREAD

**Issue by cgwalters** [bytecodealliance/rustix#215]
As you know I like a lot of the rustix design, the I/O safety, etc.  But the "no-libc" path from my use cases I feel brings more risk than reward.  One blunt way to look at this is my employer also employs people to work on glibc.  We need to support 4 different architectures.  The current status quo of using direct syscalls on e.g. x86_64 but libc on ppc64le means that suddenly my software could have ppc64le specific bugs.  (And we already have problems with CI capacity on that arch)

Basically, WDYT about exposing `use-libc` which would turn on `rustix_use_libc`?  (Or is this possible today?)

**Comment by sunfishcode** [comment]
It's possible today, by setting the `RUSTFLAGS` environment variable to `--cfg=rustix_use_libc` when building.

Also, #216 is a PR adding support for direct syscalls on ppc64le.

**Comment by cgwalters** [comment]
> It's possible today, by setting the RUSTFLAGS environment variable to --cfg=rustix_use_libc when building.

Yeah, but I want this to happen by default when one does a `cargo build` on my projects.

**Comment by sunfishcode** [comment]
I didn't do this initially because [cargo features are meant to be additive](https://doc.rust-lang.org/cargo/reference/features.html?highlight=additive#feature-unification) and at the time I didn't know of a way to model use_libc in this way. However, I now I think I can see how this could work: we could have a `use_libc` feature, and also a `use_linux_raw` feature, and require users enable at least one of those two, and if the user enables both at the same time, we'd pick one. I think that would work.

**Comment by cgwalters** [comment]
OK, thanks - I may take a swing at implementing that in the next few days.

There's obviously a much larger picture thing here about the future of rustix and the overhead of two backends, and *especially* what the default is if/when rustix is a backend of `std`.  

I've used the argument several times about Rust being more friendly to us in the "traditional Linux" space i.e. Debian/Fedora and derivatives because (unlike notably Go) it links to libc by default.  There's a lot of issues and bugs that have happened due to Go's decision, https://marcan.st/2017/12/debugging-an-evil-go-runtime-bug/ is a fun one.  But there's all the stuff around NSS modules, etc.

Anyways I won't belabor this point.  If you're good with a feature for this then I think that's great for now.
